eclipse-collections-code-generator/src/main/resources/test/list/immutable/abstractImmutablePrimitiveListTestCase.stg

```diff
@@ -288,6 +293,64 @@ public abstract class AbstractImmutable<name>ListTestCase extends AbstractImmuta
+        Immutable<name>List list2 = this.newWith(<["1"]:(literal.(type))(); separator=", ">);
+        ImmutableList\<<name><name>Pair> zipSame = list1.zip<name>(list1);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<["1", "1"]:(literal.(type))(); separator=", ">),
+                    PrimitiveTuples.pair(<["2", "2"]:(literal.(type))(); separator=", ">),
+                    PrimitiveTuples.pair(<["3", "3"]:(literal.(type))(); separator=", ">)),
+                zipSame);
+        ImmutableList\<<name><name>Pair> zipLess = list1.zip<name>(list2);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<["1", "1"]:(literal.(type))(); separator=", ">)),
+                zipLess);
+        ImmutableList\<<name><name>Pair> zipMore = list2.zip<name>(list1);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<["1", "1"]:(literal.(type))(); separator=", ">)),
+                zipMore);
+        ImmutableList\<<name><name>Pair> zipEmpty1 = list1.zip<name>(this.newWith());
+        Assert.assertTrue(zipEmpty1.isEmpty());
+        ImmutableList\<<name><name>Pair> zipEmpty2 = this.newWith().zip<name>(list1);
+        Assert.assertTrue(zipEmpty2.isEmpty());
+    }
+
+    @Test
+    public void zip()
+    {
+        Immutable<name>List list1 = this.newWith(<["1", "2", "3"]:(literal.(type))(); separator=", ">);
+        Immutable<name>List list2 = this.newWith(<["1"]:(literal.(type))(); separator=", ">);
+        ImmutableList\<String> list3 = Lists.immutable.with("1", "2", "3");
+        ImmutableList\<String> list4 = Lists.immutable.with("1");
+        ImmutableList\<<name>ObjectPair\<String\>> zipSame = list1.zip(list3);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<(literal.(type))("1")>, "1"),
+                    PrimitiveTuples.pair(<(literal.(type))("2")>, "2"),
+                    PrimitiveTuples.pair(<(literal.(type))("3")>, "3")),
+                zipSame);
+        ImmutableList\<<name>ObjectPair\<String\>> zipLess = list1.zip(list4);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<(literal.(type))("1")>, "1")),
+                zipLess);
+        ImmutableList\<<name>ObjectPair\<String\>> zipMore = list2.zip(list3);
+        Assert.assertEquals(
+                Lists.immutable.with(
+                    PrimitiveTuples.pair(<(literal.(type))("1")>, "1")),
+                zipMore);
+        ImmutableList\<<name>ObjectPair\<String\>> zipEmpty1 = list1.zip(Lists.immutable.empty());
+        Assert.assertTrue(zipEmpty1.isEmpty());
+        ImmutableList\<<name>ObjectPair\<String\>> zipEmpty2 = this.newWith().zip(Lists.immutable.with("1", "2"));
+        Assert.assertTrue(zipEmpty2.isEmpty());
```

Comment:
Tests for equal size

Reply:
Zip with the same list should cover this case.